packages/cli-kit/assets/cli-ruby/lib/shopify_cli/theme/dev_server/cdn_fonts.rb

```diff
@@ -21,40 +21,45 @@ def call(env)
 
           # Proxy the request, and replace the URLs in the response
           status, headers, body = @app.call(env)
-          body = replace_font_urls(body)
+
+          if headers["Content-Type"]&.start_with?("text/")
```

Comment:
Relevant change here.

(Other changes were result of repo/editor configuration)